### DISCUSSION TITLE
Animation?

### DISCUSSION THREAD

**Discussion by CestMoiRoma** [Povstalec/StargateJourney#125] [Q&A] [answered]
Hi! I'm new to the mod, and I knew a bit from the old SGCraft mod back in 1.7.10. In the video on CurseForge, the Milky Way gate had animation, but it doesn't anymore? In 1.20.1, only the Universe and Pegasus gates have animation? 

In any case, I really like where the mod is going and hope for the best!

**Comment by Povstalec** [comment, marked as answer]
For the Milky Way gate, there is no "animation" in the sense that you just press buttons and the gate starts rotating, like sgcraft had, since this is a completely different mod.

Instead, to make Milky Way Stargates rotate, you need to use "manual dialing":

https://github.com/Povstalec/StargateJourney/wiki/Guides#manual-dialing

**Reply by CestMoiRoma** [reply]
Oh, okay. It's just that the 'press a button and start rotating' feature works with the Universe and Pegasus, so I was wondering why it wasn't working for the Milky Way. Now that I know how to make it rotate, I'll look into the CC code you made a couple of months ago. Thanks for sharing it! :D

Edit: Just a quick question, will there be a CC block for the rings when they work? I can imagine a system like to making a keypad with a code to enable certain floors.

**Reply by Povstalec** [reply]
Yeah, eventually there will be a SGC-style dialing computer block, which will allow you to do the 'press a button and start rotating' so that people who aren't familiar with CC:Tweaked and Redstone can dial that way as well.

And yes, it will eventually be possible to control Transport Rings with CC:Tweaked as well.

**Reply by CestMoiRoma** [reply]
I think that adding the animation directly into the mod, rather than using CC would be a better approach. This would not only reduce the load on CC
but also simplify the code. Currently, I'm working on creating a screen with CC
to control the gate, but certain functions are challenging to implement due to the loops required for animating the Stargate. If the mod included functions like Interface.dial(address) that handled the animation internally, it would greatly simplify things. The logic for this is already present; it just needs to be integrated into the mod itself instead of relying on CC. I believe this integration wouldn't be too complex.

It just need to be integrated... (yet another thing to do lmao)
I'll try to look into it, but since I'm not yet familiar with Java, it might be a bit shaky at first. However, if my efforts can help the project, I'm ready to assist. I understand that projects like these can be quite complex, so any additional help could be beneficial. :)

**Reply by Povstalec** [reply]
I don't think you understood me. The SGC-style computer block will be in the mod itself, it won't be a thing limited to CC:Tweaked compatibility

**Reply by CestMoiRoma** [reply]
I understand, but if you create a block like that, all the animation code will be added, right? So why not simplify CC
by using that same code directly?

This way, you keep CC
while simplifying it at the same time. Everyone is a winner: those who want a simple block and those who prefer a more complex system with ComputerCraft. The latter can benefit from having less code to write, thanks to simpler commands.
